### PR TITLE
feature added setLabel methods to Span, Transaction of public api

### LINKED ISSUE #1446: Public API - deprecate addLabel in favor of setLabel
## Is your feature request related to a problem?

Most other agents use `setLabel` (or `SetLabel` or similar), we should align.
## Describe the solution you'd like

Add three new `setLabel` methods to the API and deprecate `addLabel`
## Describe alternatives you've considered

Allowing both

### PR DESCRIPTION
**Pull request by kananindzya**
closes elastic/apm-agent-java#1446

### CODE REVIEW

File: apm-agent-api/src/main/java/co/elastic/apm/api/Span.java [hunk 1]
```diff
@@ -162,10 +194,10 @@
      *
      * @param key   The label key.
      * @param value The label value.
-     * @since 1.5.0, APM Server 6.7
+     * @since 1.9.0, APM Server 6.7
```
Comment (SylvainJuge) [inline apm-agent-api/src/main/java/co/elastic/apm/api/Span.java:197]: [question] should it be `1.19` here for the `@since` javadoc tag instead ?

**Review by SylvainJuge — APPROVED** [review]
LGTM, I just have one question regarding the `@since` javadoc tag value.

File: apm-agent-api/src/main/java/co/elastic/apm/api/Span.java (hunk 1 above)
Comment (eyalkoren) [reply apm-agent-api/src/main/java/co/elastic/apm/api/Span.java:197]: Of course, good catch, thanks!

**Comment by felixbarny** [comment]
The APM getting started docs also mention `addLabel` so they need to be updated as well: https://www.elastic.co/guide/en/apm/get-started/current/metadata.html#labels-fields

**Pull request merged**